openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/weighting/MaximumSpeedWeighting.java

```diff
@@ -0,0 +1,120 @@
+/*  This file is part of Openrouteservice.
+ *
+ *  Openrouteservice is free software; you can redistribute it and/or modify it under the terms of the
+ *  GNU Lesser General Public License as published by the Free Software Foundation; either version 2.1
+ *  of the License, or (at your option) any later version.
+
+ *  This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Lesser General Public License for more details.
+
+ *  You should have received a copy of the GNU Lesser General Public License along with this library;
+ *  if not, see <https://www.gnu.org/licenses/>.
+ */
+package org.heigit.ors.routing.graphhopper.extensions.weighting;
+
+import com.graphhopper.routing.profiles.DecimalEncodedValue;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.routing.util.HintsMap;
+import com.graphhopper.routing.weighting.Weighting;
+import com.graphhopper.util.EdgeIteratorState;
+import com.graphhopper.util.Parameters.Routing;
+import org.heigit.ors.config.AppConfig;
+
+/**
+ * This class creates the weighting for the routing according to the maximum speed set by user.
+ *
+ * @author Athanasios Kogios
+ */
+
+public class MaximumSpeedWeighting implements Weighting {
+    protected final static double SPEED_UNIT_CONVERTER = 3.6; //From km/h to m/s.
+    private final double headingPenalty;
+    private final  double userMaxSpeed;
+    private final Weighting superWeighting;
+    private final DecimalEncodedValue avSpeedEnc;
+    private boolean calculateWeight;
+    private double maxSpeed =  ((AppConfig.getGlobal().getServiceParameter("routing.profiles.default_params","maximum_speed_lower_bound")) != null)
```

Comment:
As mentioned before, use `RouteProfileConfiguration` as a canonical place to store the dafault. `maxSpeed` could probably be passed via `HintsMap`.

Reply:
Which `HintsMap` exactly?

Reply:
Change of plans. After looking into this further I think `maximum_speed` validation should be handled even before in `RoutingProfileManager`. It should be safe to assume that once `MaximumSpeedWeighting` is created the value has been already verified so no need to check again.

Reply:
So Should I wait for further info?

Reply:
I don't have any more insights for now. My latest comments reflect how I would proceed with the implementation, however, feel free to look around yourself. Maybe there is an even better and cleaner way of doing this?